builtin/docker/platform.go

```diff
@@ -141,43 +144,80 @@ func (p *Platform) Status(
 
 	log.Debug("querying docker for container health")
 
-	resources := []*sdk.StatusReport_Resource{{
-		Name: containerInfo.Name,
-	}}
+	// Find the declared resource that corresponds to our observed container
+	var containerDeclaredResource *sdk.DeclaredResource
+	for _, r := range declaredResources.Resources {
+		if r != nil && r.Name == "container" {
+			containerDeclaredResource = r
+		}
+	}
```

Comment:
Leaving the work of mapping resources to declared resources to the plugin devs this way feels a bit clunky to me, but I think we can make this easier with ResourceManager.

Reply:
if this is going to be common, we may want to add like a `ByName()` method to `component.DeclaredResources` to facilitate this.

Reply:
Done! https://github.com/hashicorp/waypoint-plugin-sdk/commit/eaea18d2d5004fc8f8d1af4299decaa9ed643b49